### ISSUE TITLE
Visual editor formatting buttons don't wrap new text

### ISSUE THREAD

**Issue by Tarendai** [wordpress-mobile/WordPress-iOS#2862]
When selecting text and pressing bold, the strong tags surround the selected text. But if I tap bold and start typing, the text I type appears after the closing tag, not inside it, requiring me to reposition the caret

Here's a video of the process:

https://cldup.com/loacTHWSyG.mov

**Comment by rachelmcr** [comment]
I retested this issue in the visual editor in the latest beta. When the formatting button is the first thing pressed at the start of a new post, this is still an issue:

1. Open the editor (visual editor enabled in the app)
2. Press a formatting button, e.g. bold
3. Start typing

Result: The text does not have any formatting. Switching to HTML view shows there are no HTML tags in the post at all.

However, if there is already some text in the body of the post, the formatting buttons do wrap new text:

1. Open the editor (visual editor enabled in the app)
2. Enter text in the post body
3. Press a formatting button, e.g. bold
4. Start typing

Result: The new text is formatted according to the selected formatting button.

Tested on iPhone 6, iOS 9.2.1, app beta v. 5.9.0.20160126.
